src/dbnode/persist/fs/seek_manager.go

```diff
@@ -269,11 +279,58 @@ func (m *seekerManager) Return(shard uint32, start time.Time, seeker ConcurrentD
 	}
 
 	found := false
-	for i, compareSeeker := range seekersAndBloom.seekers {
+	for i, compareSeeker := range seekers.active.seekers {
 		if seeker == compareSeeker.seeker {
 			found = true
 			compareSeeker.isBorrowed = false
-			seekersAndBloom.seekers[i] = compareSeeker
+			seekers.active.seekers[i] = compareSeeker
+			break
+		}
+	}
+
+	if found {
+		return nil
+	}
+
+	// If no match was found in the active seekers, it's possible that an inactive seeker is being returned.
+	for i, compareSeeker := range seekers.inactive.seekers {
+		if seeker == compareSeeker.seeker {
+			found = true
+			compareSeeker.isBorrowed = false
+			seekers.inactive.seekers[i] = compareSeeker
+
+			// The goroutine that returns the last outstanding inactive seeker is responsible for notifying any
+			// goroutines waiting for all inactive seekers to be returned and clearing out the inactive seekers
+			// state entirely.
+			allAreReturned := true
+			for _, inactiveSeeker := range seekers.inactive.seekers {
+				if inactiveSeeker.isBorrowed {
+					allAreReturned = false
+					break
+				}
+			}
+
+			if !allAreReturned {
+				break
+			}
+
+			// All the inactive seekers have been returned so it's safe to signal and clear them out.
+			var multiErr = xerrors.NewMultiError()
```

Comment:
super nit: More idiomatic to just use `:=` instead of untyped var `var foo = bar`